pkg/devfile/parse.go

```diff
@@ -39,12 +39,6 @@ func Parse(path string) (d DevfileObj, err error) {
 		return d, errors.Wrapf(err, "failed to decode devfile content")
 	}
 
-	// Validate devfile data
-	err = d.Data.Validate()
```

Comment:
Which packages are considered independent of odo? Is it `pkg/devfile/parser` and `pkg/devfile/versions` only or were you thinking more? If it's only those two then can we move the existing validation files to a new package like `pkg/devfile/validation` instead of removing them? I'm thinking for odo we'd want the `pkg/devfile/Parse` function to still validate here but just call the validation from an odo package.

Perhaps we should consider moving the parser packages to a top-level folder in pkg as well? Maybe `pkg/parser` or `pkg/devfile-parser` so it will make it easier to separate it out later?